module/zfs/arc.c

```diff
@@ -4700,10 +4706,17 @@ arc_free_memory(void)
 	return (ptob(si.freeram - si.freehigh));
 #else
 #ifdef ZFS_GLOBAL_NODE_PAGE_STATE
+#ifdef ZFS_GLOBAL_ZONE_PAGE_STATE
+	return (ptob(nr_free_pages() +
+	    global_node_page_state(NR_INACTIVE_FILE) +
+	    global_node_page_state(NR_INACTIVE_ANON) +
+	    global_zone_page_state(NR_SLAB_RECLAIMABLE)));
+#else
 	return (ptob(nr_free_pages() +
 	    global_node_page_state(NR_INACTIVE_FILE) +
 	    global_node_page_state(NR_INACTIVE_ANON) +
-	    global_node_page_state(NR_SLAB_RECLAIMABLE)));
+	    global_page_state(NR_SLAB_RECLAIMABLE)));
+#endif /* ZFS_GLOBAL_ZONE_PAGE_STATE */
```

Comment:
Please triple check this but I believe we should be instead checking for the change of NR_SLAB_RECLAIMABLE from a zone_stat_item to a node_stat_item.  In fact we never need to use the `global_zone_page_state` interface since by the time it was added `NR_SLAB_RECLAIMABLE` was moved to the node.

```c
#if defined(ZFS_GLOBAL_NODE_PAGE_STATE) && \
    defined(ZFS_NODE_STAT_NR_SLAB_RECLAIMABLE)
        /* 4.13+ API */
        return (ptob(nr_free_pages() +
            global_node_page_state(NR_INACTIVE_FILE) +
            global_node_page_state(NR_INACTIVE_ANON) +
            global_node_page_state(NR_SLAB_RECLAIMABLE)));
#elif defined(ZFS_GLOBAL_NODE_PAGE_STATE)
        /* 4.9 - 4.12 API */
        return (ptob(nr_free_pages() +
            global_node_page_state(NR_INACTIVE_FILE) +
            global_node_page_state(NR_INACTIVE_ANON) +
            global_page_state(NR_SLAB_RECLAIMABLE)));
#else
        /* x - 4.8 API */
        return (ptob(nr_free_pages() +
            global_page_state(NR_INACTIVE_FILE) +
            global_page_state(NR_INACTIVE_ANON) +
            global_page_state(NR_SLAB_RECLAIMABLE)));
#endif
```

Reply:
@behlendorf That looks correct to me.